metadata-ingestion/src/datahub/ingestion/source/unity/source.py

```diff
@@ -694,14 +719,26 @@ def _generate_lineage_aspect(
                 UpstreamClass(
                     dataset=upstream_urn,
                     type=DatasetLineageTypeClass.TRANSFORMED,
+                    created=AuditStampClass(
+                        time=int(upstream_ref.last_updated.timestamp() * 1000)
+                        if upstream_ref.last_updated
+                        else int(time.time() * 1000),
+                        actor="urn:li:corpuser:unknown",
+                    ),
                 )
             )
 
-        for notebook in table.upstream_notebooks:
+        for notebook in table.upstream_notebooks.values():
             upstreams.append(
                 UpstreamClass(
-                    dataset=self.gen_notebook_urn(notebook),
+                    dataset=self.gen_notebook_urn(notebook.id),
                     type=DatasetLineageTypeClass.TRANSFORMED,
+                    created=AuditStampClass(
```

Comment:
use auditstamp instead of created

Reply:
also we should not be falling back to time.time() (unless this is a required field?)

but actually if `notebook.last_updated is None`, then we shouldn't even be setting a created/audit stamp